### PR TITLE
Add junit report for test targets

### PR DESCRIPTION
**Pull request by periklis**
### Description
This PR adds junit report generation for the `test-unit` target.

/cc @blockloop 
/assign @ewolinetz @jcantrill 

#### JUnit Report for Unit Tests:
image

e.g. https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_elasticsearch-operator/564/pull-ci-openshift-elasticsearch-operator-master-unit/1324361140788006912

##### Bash Tests
image

##### Golang Tests
image

e.g. https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_elasticsearch-operator/564/pull-ci-openshift-elasticsearch-operator-master-e2e-operator/1324361140477628416

### Links

- JIRA: https://issues.redhat.com/browse/LOG-976

### CODE REVIEW

**Review by jcantrill — COMMENTED** [review]
/approve

File: Makefile [hunk 1]
```diff
@@ -28,6 +29,9 @@ OS_NAME=$(shell uname -s | tr '[:upper:]' '[:lower:]')
 
 all: build
 
+artifactdir:
```
Comment (blockloop) [inline Makefile:32]: mark this as `.PHONY`

**Review by blockloop — APPROVED** [review]
/lgtm

**Comment by periklis** [comment]
@blockloop PTAL again. It is now good for me.

File: .bingo/Variables.mk [hunk 2]
```diff
@@ -22,6 +22,12 @@ $(BINGO): .bingo/bingo.mod
 	@echo "(re)installing $(GOBIN)/bingo-v0.2.3"
 	@cd .bingo && $(GO) build -mod=mod -modfile=bingo.mod -o=$(GOBIN)/bingo-v0.2.3 "github.com/bwplotka/bingo"
 
+GO_JUNIT_REPORT := $(GOBIN)/go-junit-report-v0.9.1
```
Comment (blockloop) [inline .bingo/Variables.mk:25]: Shouldn't you use `$(GO_JUNIT_REPORT)` instead of `$(GOBIN)/go-junit-report-v0.9.1`

**Pull request merged**